core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1599,4 +1628,18 @@ object KafkaZkClient {
       time, metricGroup, metricType)
     new KafkaZkClient(zooKeeperClient, isSecure, time)
   }
+
+
+  def controllerZkVersionCheck(version: Int): Option[ZkVersionCheck] = {
+    if (version < 0)
+      None
+    else
+      Some(ZkVersionCheck(ControllerEpochZNode.path, version))
+  }
+
+  def maybeThrowControllerMoveException(response: AsyncResponse): Unit = {
```

Comment:
Given that there may be other zookeeper operation other than `controllerZkVersionCheck` which can also check the zkversion of the corresponding znode, some response may also show `zkVersionCheckResultCode != Code.OK` and cause `maybeThrowControllerMoveException` to throw `ControllerMovedException` even if it is not for the `controllerZkVersionCheck`. Will this be a problem?

Also, any chance we can also include the expected zkversion in the message of `ControllerMovedException`?

Reply:
Yes, I think it is better to include the expected zkVersion in the zookeeper response and extract the information from the response when throwing `ControllerMovedException`. I have added `zkVersionCheckResult: Option[ZkVersionCheckResult]` to achieve this.